Cesil/Configuration/Options.cs

```diff
@@ -170,6 +174,11 @@ public sealed class Options : IEquatable<Options>
         /// How to handle extra colums encountered when reading a CSV.
         /// </summary>
         public ExtraColumnTreatment ExtraColumnTreatment { get; private set; }
+        /// <summary>
+        /// Which ArrayPool to use when buffering DynamicCellValues when
+        /// writing dynamic rows.
+        /// </summary>
+        public ArrayPool<DynamicCellValue> ArrayPool { get; private set; }
```

Comment:
This array pool is specifically the array pool used for writing dynamic rows, so `ArrayPool` is perhaps not the best name for it. I can envision you wanting to add other configurable array pools

Reply:
```csharp
interface IArrayPoolFactory
{
    ArrayPool<T> Create<T>();
}
class DefaultArrayPoolFactory : IArrayPoolFactory
{
    public ArrayPool<T> Create<T>() => ArrayPool<T>.Shared;
}

class Options
{
    // ...
    public IArrayPoolFactory ArrayPoolFactory { get; }
}
```

naming debatable - perhaps you don't want to imply that the method will always create an array pool

Reply:
Good idea, an interface for obtaining arbitrary pools would be a more flexible solution than hard coding the ones needed right now by Cesil.

Per https://github.com/kevin-montrose/Cesil/issues/6#issuecomment-650771069 I'm thinking it'll actually hand back `MemoryPool<T>`.  Also need to think about provided contexts... I could imagine wanting to dedicate particular pools to particular `IBoundConfiguration<T>` instances...

Reply:
Convinced myself that a context object isn't necessary, since `MemoryPool<T>`s are acquired during `Configuration.ForXXX()` calls - so if you really care to tie a specific `IMemoryPoolProvider`-sourced `MemoryPool<T>` to an `IBoundConfiguration<T>` you can just do it via the `Options` passed.